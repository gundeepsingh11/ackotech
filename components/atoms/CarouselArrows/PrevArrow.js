// @flow
/**
 * PrevArrow component to represent the left arrow for carousel
 * @module PrevArrow
 */
import React, { Fragment } from 'react';
import withStyles from '../../../lib/withStyles';
import styles from './Arrow.style';

const PrevArrow = ({ onClick, className }) => (
  <Fragment>
    <button
      onClick={onClick}
      className={`icon-left-arrow-large arrow-small ${className}`}
      type="button"
    />
  </Fragment>
);

export default withStyles(PrevArrow, styles);
export { PrevArrow as PrevArrowVanilla };
