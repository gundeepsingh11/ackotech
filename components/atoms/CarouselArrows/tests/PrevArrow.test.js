import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { PrevArrowVanilla } from '../PrevArrow';

const props = {
  onClick: jest.fn(),
  className: 'arrow',
};

describe('<PrevArrow />', () => {
  let Component = '';
  beforeEach(() => {
    Component = shallow(<PrevArrowVanilla {...props} />);
  });

  test('should render correctly', () => {
    expect(Component).toMatchSnapshot();
  });

  test('should render with styles correctly and match snapshot', () => {
    Component = renderer.create(<PrevArrowVanilla />).toJSON();
    expect(Component).toMatchSnapshot();
  });
});
