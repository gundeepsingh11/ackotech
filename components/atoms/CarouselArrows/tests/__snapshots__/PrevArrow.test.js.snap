// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrevArrow /> should render correctly 1`] = `
<Fragment>
  <button
    className="icon-left-arrow-large arrow-small arrow"
    onClick={[MockFunction]}
    type="button"
  />
</Fragment>
`;

exports[`<PrevArrow /> should render with styles correctly and match snapshot 1`] = `
<button
  className="icon-left-arrow-large arrow-small undefined"
  type="button"
/>
`;
